app/controllers/profile/users_controller.rb

```diff
@@ -9,15 +9,18 @@ def edit
 
   def update
     @user = current_user
-    @user.update(user_params)
-    flash[:success] = "Your data is updated"
-    redirect_to profile_path
+    if @user.update(user_params)
+      flash[:success] = "Your data is updated"
+      redirect_to profile_path
+    else
+      flash[:error] = "the email you entered is already taken"
+      render :edit
+    end
   end
 
   private
 
   def user_params
-    result = params.require(:user).permit(:name, :address, :city, :state, :email, :password, :zip_code, :password_confirmation)
-    result.reject{|key, value| value.empty?}
```

Comment:
Is there a reason we're deleting this line?

Reply:
I have no idea how that got deleted. I did not do it.